tensorly/decomposition/_tucker.py

```diff
@@ -294,7 +349,156 @@ def non_negative_tucker(tensor, rank, n_iter_max=10, init='svd', tol=10e-5,
                 print('converged in {} iterations.'.format(iteration))
             break
 
-    return TuckerTensor((nn_core, nn_factors))
+    tensor = TuckerTensor((nn_core, nn_factors))
+    if return_errors:
+        return tensor, rec_errors
+    else:
+        return tensor
+    
+def non_negative_tucker_hals(tensor, rank, n_iter_max=100, init="svd", svd='numpy_svd', tol=1e-8,
+                             sparsity_coefficients=None, fixed_modes=None, random_state=None,
+                             verbose=False, normalize_factors=False, return_errors=False, exact=False, 
+                             algorithm='fista'):
+    """
+    Non-negative Tucker decomposition
+
+    Uses HALS to update each factor columnwise and uses
+    fista or active set algorithm to update the core, see [1]_ 
+    
+    Parameters
+    ----------
+    tensor : ndarray
+    rank   : int
+            number of components
+    n_iter_max : int
+            maximum number of iteration
+    init : {'svd', 'random'}, optional
+    svd : str, default is 'numpy_svd'
+        function to use to compute the SVD, acceptable values in tensorly.SVD_FUNS
+    tol : float, optional
+          tolerance: the algorithm stops when the variation in
+          the reconstruction error is less than the tolerance
+        Default: 1e-8
+    sparsity_coefficients : array of float (as much as the number of modes)
+        The sparsity coefficients are used for each factor
+        If algorithm is fista, last coefficient is used when updating core
+        If set to None, the algorithm is computed without sparsity
+        Default: None
+    fixed_modes : array of integers (between 0 and the number of modes)
+        Has to be set not to update a factor, 0 and 1 for U and V respectively
+        Default: None
+    verbose : boolean
+        Indicates whether the algorithm prints the successive
+        reconstruction errors or not
+        Default: False
+    return_errors : boolean
+        Indicates whether the algorithm should return all reconstruction errors
+        and computation time of each iteration or not
+        Default: False
+    exact : If it is True, the HALS nnls subroutines give results with high precision but it needs high computational cost. 
+        If it is False, the algorithm gives an approximate solution.
+        Default: False
+    algorithm : {'fista', 'as'}
```

Comment:
I would explicitly use `'active_set'` instead of just `'as'`

Reply:
It is 'active_set' now.